### PR TITLE
Ty/offers api

### PR DESCRIPTION
**Pull request by tylerjnap**
- Added `Offers` resource
- Updated change log, moving `'Added CODEOWNERS'` to v 2.4.0 (think there was mistake with adding to the CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 Current Version
-  - Added CODEOWNERS
+  - Add `Offers` resource
 
 2.4.0 December 6, 2017
   - Add `Links` resource
+  - Added CODEOWNERS
```
Comment (griffinmyers) [inline CHANGELOG.md]: It looks like this wasn't a part of 2.4.0, it was after.

File: lib/button/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Button
-  VERSION = '2.4.0'.freeze
+  VERSION = '2.5.0'.freeze
```
Comment (griffinmyers) [inline lib/button/version.rb]: Mind backing this out to `2.4.0`?  I can cut a release in another change if you'd like.

File: README.md [hunk 3]
```diff
@@ -281,6 +282,24 @@ puts response
...
+
+##### Get Offers
+
+```ruby
+require 'button'
+
+client = Button::Client.new('sk-XXX')
+
+response = client.offers.get_offers({
+    user_id: "some-user-id",
```
Comment (griffinmyers) [inline README.md]: Is this the Publisher's id of the user?

File: README.md [hunk 4]
```diff
@@ -281,6 +282,24 @@ puts response
...
+    device_ids: ["123"]
```
Comment (griffinmyers) [inline README.md]: What kind of device ids get submitted here?

File: lib/button/resources/offers.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    def path
+      '/v1/offers'
+    end
+
+    # Retrieve offers that are available to Publishers user
+    #
+    # @param [Hash] user to retrieve offers for
+    # @return [Button::Response] the API response
+    #
+    def get_offers(user)
```
Comment (griffinmyers) [inline lib/button/resources/offers.rb]: nit: `get_offers` seems a bit redundant when you pair it with the namespace.  A few suggestions if you're open to it:

```ruby
client.offers.get(user_id: '123') # ?
client.offers.get_by_user(user_id: '123') # ?
```

**Review by griffinmyers — CHANGES_REQUESTED** [review]
Looks generally awesome, thanks for submitting!

Only blocking change is straightening about the version comments.

File: CHANGELOG.md (hunk 1 above)
Comment (tylerjnap) [reply CHANGELOG.md]: I think there was an error in the CHANGELOG.

It was in `Current Version` which would in the current state is 2.5.0, even though this release would be be 2.5.0.

Happy to arrange this as you see fit

File: lib/button/version.rb (hunk 2 above)
Comment (tylerjnap) [reply lib/button/version.rb]: Sure 👍

File: README.md (hunk 3 above)
Comment (tylerjnap) [reply README.md]: Yes. All parameters are [here](https://developer.usebutton.com/api#offers)

File: README.md (hunk 4 above)
Comment (tylerjnap) [reply README.md]: Same as above ^^

File: lib/button/resources/offers.rb (hunk 5 above)
Comment (tylerjnap) [reply lib/button/resources/offers.rb]: 👍

**Review by griffinmyers — APPROVED** [review]

**Review by tadasv — APPROVED** [review]

**Pull request merged**